### PR TITLE
SNOW-1442354 Fix a bug in series.reset_index where the result name may be wrong

### PR DESCRIPTION
**Pull request by sfc-gh-azhan**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1442354 Fix a bug in series.reset_index where the result name may be wrong

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Review by sfc-gh-nkrishna — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 1]
```diff
@@ -78,7 +78,8 @@
 - Fixed AssertionError in `Series.drop_duplicates` and `DataFrame.drop_duplicates` when called after `sort_values`.
 - Fixed a bug in `Index.to_frame` where the result frame's column name may be wrong where name is unspecified.  
 - Fixed a bug where some Index docstrings are ignored. 
-
+- Fixed a bug in `Series.reset_index(drop=True)` where the result name may be wrong.
```
Comment (sfc-gh-nkumar) [inline CHANGELOG.md:81]: I this is bit broad. Maybe consider changing to `Fixed a bug in ... where index column name in result maybe wrong.

**Review by sfc-gh-nkumar — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (sfc-gh-azhan) [reply CHANGELOG.md:81]: It's not the index column name, It return a series so the result name mean the series name was wrong.

File: src/snowflake/snowpark/modin/pandas/series.py [hunk 2]
```diff
@@ -1833,7 +1833,9 @@ def reset_index(
         Generate a new Series with the index reset.
         """
         # TODO: SNOW-1063347: Modin upgrade - modin.pandas.Series functions
-        if name is no_default:
+        if drop:
```
Comment (sfc-gh-azhan) [inline src/snowflake/snowpark/modin/pandas/series.py:1836]: @sfc-gh-joshi I made a frontend change.

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**